static/app/components/charts/intervalSelector.spec.tsx

```diff
@@ -0,0 +1,80 @@
+import {render} from 'sentry-test/reactTestingLibrary';
+
+import IntervalSelector from 'sentry/components/charts/intervalSelector';
+import EventView from 'sentry/utils/discover/eventView';
+
+describe('IntervalSelector', function () {
+  const project = TestStubs.Project();
+  const eventView = EventView.fromSavedQuery({
+    id: '',
+    name: 'test query',
+    version: 2,
+    fields: ['transaction', 'count()'],
+    projects: [project.id],
+  });
+  it('increases small interval', function () {
+    let interval = '1s';
+    eventView.interval = interval;
+    eventView.statsPeriod = '90d';
+    const intervalSelector = (
+      <IntervalSelector
+        eventView={eventView}
+        onIntervalChange={newInterval => (interval = newInterval)}
+      />
+    );
+    render(intervalSelector);
+    expect(interval).toEqual('4h');
+  });
+  it('reducing uses clean day shorthands', function () {
+    let interval = '2d';
+    eventView.interval = interval;
+    eventView.statsPeriod = interval;
+    const intervalSelector = (
+      <IntervalSelector
+        eventView={eventView}
+        onIntervalChange={newInterval => (interval = newInterval)}
+      />
+    );
+    render(intervalSelector);
+    expect(interval).toEqual('1d');
+  });
+  it('reducing uses clean hour shorthands', function () {
+    let interval = '2h';
+    eventView.interval = interval;
+    eventView.statsPeriod = interval;
+    const intervalSelector = (
+      <IntervalSelector
+        eventView={eventView}
+        onIntervalChange={newInterval => (interval = newInterval)}
+      />
+    );
+    render(intervalSelector);
+    expect(interval).toEqual('1h');
```

Comment:
This seems to be duplicated multiple times, perhaps we should just have a common function each test case

Reply:
:+1: I'll do this as I add more tests